lib/OpenQA/Schema/Result/JobGroups.pm

```diff
@@ -382,68 +386,91 @@ sub template_data_from_yaml {
-                    my $job_template_key
-                      = $arch . $product_name . $machine_name . ($testsuite_name // '') . ($job_template_name // '');
-                    if ($job_template_names{$job_template_key}) {
-                        my $name = $job_template_name // $testsuite_name;
-                        my $error = "Job template name '$name' is defined more than once. "
-                          . "Use a unique name and specify 'testsuite' to re-use test suites in multiple scenarios.";
-                        return {error => $error};
-                    }
-                    $job_template_names{$job_template_key} = {
-                        prio => $prio,
-                        machine_name => $machine_name,
-                        arch => $arch,
-                        product_name => $product_name,
-                        product_spec => $yaml_products->{$product_name},
-                        job_template_name => $job_template_name,
-                        testsuite_name => $testsuite_name,
-                        settings => $settings,
-                        length $description ? (description => $description) : (),
-                    };
+                if (exists $attr->{testsuite}) {
+                    $job_template_name = $testsuite_name;
+                    $testsuite_name = $attr->{testsuite};
                 }
+                if ($attr->{settings}) {
+                    %$settings = (%{$settings // {}}, %{$attr->{settings}});
+                }
+                if (defined $attr->{description}) {
+                    $description = $attr->{description};
+                }
+            }
+            else {
+                $testsuite_name = $spec;
             }
+
+            $machine_names = [$machine_names] if ref($machine_names) ne 'ARRAY';
+            _parse_job_template_machines($machine_names,
+                $job_template_names,
+                $prio,
+                $arch,
+                $product_name,
+                $yaml_products,
+                $job_template_name,
+                $testsuite_name,
+                $settings,
+                $description);
         }
     }
+}
 
-    return \%job_template_names;
+sub _parse_job_template_machines ($machine_names, $job_template_names, $prio, $arch, $product_name, $yaml_products, $job_template_name, $testsuite_name, $settings, $description) {
+    # my ($machine_names, %job_template_names, $prio, $arch, $product_name, $yaml_products, $job_template_name, $testsuite_name, $settings, $description) = @_;
+    foreach my $machine_name (@{$machine_names}) {
+        my $job_template_key
+          = $arch . $product_name . $machine_name . ($testsuite_name // '') . ($job_template_name // '');
+        if ($job_template_names->{$job_template_key}) {
+            my $name = $job_template_name // $testsuite_name;
+            my $error = "Job template name '$name' is defined more than once. "
```

Comment:
The `to_yaml` method was previously returning this error.
Now you are returning this in a nested function, so you would need to catch it where you call it and pass it on.
For that exceptions are good., e.g. `die` and then catch it with `eval` or `try/catch`.
You could move this change out of the PR, so that the rest can be approved and merged.

Reply:
You just pushed a change a minute after my comment that fixes it. Since this is always returning a hash ref and passing it on, the `die` approach is not needed.